internal/ociregistry/server.go

```diff
@@ -255,12 +255,10 @@ func (s *Server) procureToken(hdr string) error {
 
 	err = json.NewDecoder(resp.Body).Decode(&tokenBody)
 	if err != nil {
-		return errors.Wrapf(err, "error decoding www-authenticate header")
+		return "", errors.Wrapf(err, "error decoding www-authenticate header")
 	}
 
-	s.Auth = fmt.Sprintf("%s %s", authType, tokenBody.Token)
-
-	return nil
+	return fmt.Sprintf("%s %s", authType, tokenBody.Token), nil
 }
 
 func (s *Server) Negotiate() error {
```

Comment:
Do we even need the exported `Negotiate` and to call that from plugins anymore then? Or does this happen automatically on every API call anyways?

Reply:
We do call it from all the plugins.

Reply:
For example, https://github.com/hashicorp/waypoint/pull/2056/commits/4bb27782c706bf9b98a0ac79a521d48e8691f6fc#diff-678d37d13f4a054528633c9b23c7f4161cd73755c1b3145cb65e616b7fc25c72R84. I modified it in 4bb to take the repo to setup the transport.